dashboard/src/components/Config/AppRepoList/AppRepoControl.tsx

```diff
@@ -33,19 +32,8 @@ export function AppRepoControl({
   const handleDeleteClick = (repoName: string, repoNamespace: string, currentNamespace: string) => {
     return async () => {
       await dispatch(actions.repos.deleteRepo(repoName, repoNamespace));
-      if (currentNamespace !== kubeappsNamespace) {
-        // Re-fetch repos in both namespaces because otherwise, the state
-        // will be updated only with the repos of repoNamespace and removing
-        // the global ones.
-        // TODO(andresmgot): This can be refactored once hex UI is dropped
-        dispatch(actions.repos.fetchRepos(currentNamespace, kubeappsNamespace));
-      } else {
-        if (currentNamespace === definedNamespaces.all || currentNamespace === kubeappsNamespace) {
-          dispatch(actions.repos.fetchRepos(currentNamespace));
-        } else {
-          dispatch(actions.repos.fetchRepos(currentNamespace, kubeappsNamespace));
-        }
-      }
+      // repoNamespace?
```

Comment:
Yes, this is related to my earlier question: I guess the scenario you're questioning here is like the one I described, except that in this case you delete a repo whic could have been from the current namespace or from another. I assume we'd need to refetch repos from the namespace of the deleted repo?

Reply:
Yes, I added the comment to not forget about that (and simplify this implementation).

Reply:
Yep, but shouldn't the following line use `repoNamespace` instead of `currentNamespace`, since that's the list of repos which will have changed?

Reply:
I had to re-check this. No, if we are in the "all namespaces" scenario, we should delete the repo from its namespace but then we need to fech repos from all namespaces (`currentNamespace` is an empty string in that scenario).

Reply:
Ah - so perhaps a comment that "currentNamespace will either be the same as repoNamespace or empty if browsing across all namespaces".